feat(network): Autocertifier network integration WIP

juslesan opened the pull request:
## Summary

Use the autocertifier to create domain names and TLS certificates for nodes running WSS servers.

juslesan committed:
circular dependency in monorepo

ptesavol committed:
resolved build conflict

juslesan committed:
fix monorepo package.json package ordering

juslesan committed:
checkpoint

juslesan committed:
checkpoint

juslesan committed:
connection to server works, check connectivity

juslesan committed:
add fake peerdescriptor

juslesan committed:
add logging

juslesan committed:
add logging

juslesan committed:
add logging

juslesan committed:
isOnwPeerDescriptor return false if undefined

juslesan committed:
move logic

juslesan committed:
revert check

juslesan committed:
add logging

juslesan committed:
bind

ptesavol committed:
fix: fixed waiting for more than 32 bit milliseconds

juslesan committed:
Merge branch 'autocertifier-in-network' into autocertifier-in-network-wip

juslesan committed:
add error log

juslesan committed:
logging

juslesan committed:
add logging

juslesan committed:
logging

juslesan committed:
logging

ptesavol committed:
add production test

ptesavol committed:
convert path to node format

ptesavol committed:
added logging

ptesavol committed:
add fqdn as parameter for self-signed cert generation

ptesavol committed:
fix async await

Commit:
local change

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
added fqdn to altnames

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
increase test timeout

ptesavol committed:
fixed test

ptesavol committed:
merged

ptesavol committed:
improved test

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
added test for case where subdomain exists

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
encode uri components

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
increase logging

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
increase logging

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
MORE LOGGING

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
increase logging

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
bugfix

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
decrease logging

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

ptesavol committed:
fix test

Commit:
Merge branch 'autocertifier-in-network' of github.com:streamr-dev/network into autocertifier-in-network

Commit:
give more port options

juslesan committed:
Merge branch 'autocertifier-in-network' into autocertifier-in-network-wip

juslesan committed:
checkpoint

juslesan committed:
checkpoint, entry point cannoct open ws connections?

juslesan committed:
remove console.log

juslesan committed:
increase test timeouts

juslesan committed:
checkpoint: making browser tests pass

juslesan committed:
browser tests pass using clear text ws. Autocertify WebSocket server by default

juslesan committed:
add run script

juslesan committed:
add logging

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network

juslesan committed:
Merge branch 'autocertifier-in-network' into autocertifier-in-network-wip

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network

juslesan committed:
Merge branch 'autocertifier-in-network' into autocertifier-in-network-wip

juslesan committed:
checkpoint

juslesan committed:
try end-to-end

juslesan committed:
update certificate

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network

juslesan committed:
Merge branch 'autocertifier-in-network' into autocertifier-in-network-wip

juslesan committed:
selfSigned use rejectUnauthorized

juslesan committed:
autocertifier testing

juslesan committed:
fix build

juslesan committed:
waitForEvent3

juslesan committed:
increase timeout

juslesan committed:
add wss to StreamrChallenger

juslesan committed:
add selfSigned true flag

juslesan committed:
add logging

juslesan committed:
add ip to altNames

juslesan committed:
logging

juslesan committed:
pass altNames

juslesan committed:
add full error log

juslesan committed:
try to log subprobelms

juslesan committed:
JSON.stringy error, try adding fqdn to altNames

juslesan committed:
logging

juslesan committed:
pad altNames

juslesan committed:
remove altNames option

juslesan committed:
setHost

juslesan committed:
handle invalid DNS questions

juslesan committed:
happy path works in prod testing. All tests pass using websocketServerEnabled: false

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/streamr-dev/network?pullRequest=1975) <br/>All committers have signed the CLA.

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network-wip

juslesan committed:
fix build, eslint, versions

juslesan committed:
versions

juslesan committed:
eslint

juslesan committed:
add comment

juslesan committed:
fix config

juslesan committed:
fix test

juslesan committed:
autocertifier-client false

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network-wip

juslesan committed:
package-lock

juslesan committed:
remove node: prefix from buffer in webpack

juslesan committed:
Merge branch 'NET-1041-autocertifier' into autocertifier-in-network-wip

juslesan committed:
refactor(DHT): Connectivity checking now uses a for loop for reattempts (#1982)

## Summary

Refactored the connectivity checker to use a for loop for reattempts.
Improved error logging to include information of the entry points
PeerDescriptor

juslesan committed:
fix cli-tools tests

juslesan committed:
clean

juslesan committed:
clean

juslesan committed:
broker test autocertifier

juslesan committed:
use rejectUnauthorized: false instead of CA cert

juslesan committed:
configurability

juslesan committed:
autocertifier ci tests

teogeb committed:
refactor(trackerless-network): Rename `NetworkRpc` (#1979)

Renamed `NetworkRpc` service to `DeliveryRpc`

And renamed related RPC classes:
- `DeliveryRpcRemote` (was `RemoteRandomGraphNode`)
- `DeliveryRpcLocal` (was `StreamNodeServer`)

teogeb committed:
refactor: use `forEach` (#1980)

Use `forEach` instead of `map` in cases where we just iterate an array without mapping the items.

Also removed obsolete `DhtNode#getOpenInternetPeerDescriptors`

juslesan committed:
fix ConfigTest

juslesan committed:
sad path handling

juslesan committed:
reduce log level

juslesan committed:
feat(DHT): NET-1136 maxMessageSize configuration. (#1984)

## Summary

Added `maxMessageSize` configuration for WebSocket and WebRTC
connections.

## Changes

- Parameters previously passed to `WebSocketServer#start` are now passed
in the constructor.
- Added `WebSocketConnectorConfig` to pass parameter values to the
WebSocketConnector's contructor

juslesan committed:
Merge branch 'streamr-1.0' into autocertifier-in-network-wip

juslesan committed:
remove comment

juslesan committed:
AutoCertifierClientFacade

juslesan committed:
rename

juslesan committed:
configurability for subdomainFilePath

juslesan committed:
comments

juslesan committed:
clean

juslesan committed:
clean

juslesan committed:
load env variables with dotenv

juslesan committed:
remove comment

juslesan committed:
update readme

juslesan committed:
fix package.json

juslesan committed:
debug logging

juslesan committed:
fix client config

juslesan committed:
use arrow functions

juslesan committed:
create directory if does not exist

juslesan committed:
eslint

juslesan committed:
remove obsolete start method

juslesan committed:
eslint

juslesan committed:
eslint

juslesan committed:
remove space

juslesan committed:
env variable naming
